databases/week3/homework3.sql

```diff
@@ -0,0 +1,135 @@
+ title varchar(255),
+ description text,
+ meal_id int unsigned not null,
+ stars int,
+ created_date date,
+ FOREIGN KEY (meal_id) REFERENCES meal(id) ON UPDATE CASCADE
+);
+-- Get all meals
+select * from meal;
+-- Add a new meal
+insert into meal(title, description, location, when_, max_reservations, price, created_date)
+values('dolma', 'iraqi traditional food', 'copenhagen', '2020-08-20', '5', '200' , '2019-08-18');
+insert into meal(title, description, location, when_, max_reservations, price, created_date)
+values('flafel', 'iraqi traditional food', 'copenhagen', '2020-08-20', '6', '50' , '2020-08-30');
+insert into meal(title, description, location, when_, max_reservations, price, created_date)
+values('Rød grød med fløde', 'mexixan traditional food', 'copenhagen', '2020-08-20', '3', '100' , '2020-08-17');
+-- Get a meal with any id
+select * from meal
+where id=2;
+-- Update a meal with any id
+update meal 
+set title='shawrma', max_reservations='4'
+where id=3;
+-- Delete a meal with any id
+delete from meal
+where id=1;
+
+-- Get all reservations
+select * from reservation;
+-- Add a new reservation
+SET FOREIGN_KEY_CHECKS=0;
+insert into reservation( number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+values(3, 1, '2018-05-15', '07777777', 'nuh kjikjs', 'abs@gmail.com');
+insert into reservation( number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+values(5, 2, '2018-05-15', '077755577', 'kjikjs', 'abs@gmail.com');
+insert into reservation( number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email)
+values(3, 3, '2018-05-15', '0126564777', 'nuh kjjhhkjs', 'abs@gmail.com');
+-- Get a reservation with any id
+select * from reservation
+where id=2;
+-- Update a reservation with any id
+update reservation 
+set contact_name='abed hud', contact_email='gts@gmail.com'
+where id=3;
+-- Delete a reservation with any id
+delete from reservation
+where id=1;
+
+-- Get all reviews
+select * from review;
+-- Add a new review
+SET FOREIGN_KEY_CHECKS=0;
+insert into review( title, description, meal_id, stars, created_date)
+values('good', 'so yummy food', 3, 4, '2018-06-12');
+insert into review( title, description, meal_id, stars, created_date)
+values('bad', 'good food', 2, 8, '2018-06-12');
+insert into review( title, description, meal_id, stars, created_date)
+values('good', 'wow food', 1, 5, '2018-06-12');
+-- Get a review with any id
+select * from review
+where id=3;
+-- Update a review with any id
+update review 
+set tilte='fgdyj', stars='8'
+where id=2;
+-- Delete a review with any id
+delete from review
+where id=1;
+
+-- Get meals that has a price smaller than a specific price
+select * from meal
+where price <=100;
+
+-- Get meals that still has available reservations
+select *
+from meal
+left join reservation on meal.id=reservation.meal_id
+where meal.max_reservations > reservation.number_of_guests
+group by meal.id;
+
```

Comment:
Here you need to use the `sum(reservations.number_of_guests)` 
It need a little more conditions which works with the date of reservations.
In the current case you will soon run out of meals.

Reply:
please , can u explain more why should i use date of reservations? and i already have column called created_date what will be the difference between them?

Reply:
Hello
You have the created_date which is the date that the reservation was made and you also have a date for the reservation.
It is an extra tweak. Not part of the assignment but it will give your work more sense.